cmd/agent/app/integrations.go

```diff
@@ -345,7 +345,7 @@ func install(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("could not get current version of %s: %v", integration, err)
 	}
 
-	if versionToInstall.Equal(*currentVersion) {
+	if currentVersion != nil && versionToInstall.Equal(*currentVersion) {
```

Comment:
Seems that the only case where we can reach this line with a nil `currentVersion` is due to this case:
https://github.com/DataDog/datadog-agent/blob/4dad3b9625384667c6e667ee05ac9e4a60dc8e6b/cmd/agent/app/integrations.go#L655-L658

Should we return a `nil, someError` instead of `nil, nil` in the case quoted above? or there is a reason for returning `nil, nil`?

Reply:
Yes there is. If you look at the Python code that's executed here https://github.com/DataDog/datadog-agent/blob/4dad3b9625384667c6e667ee05ac9e4a60dc8e6b/cmd/agent/app/integrations.go#L45 you see that we expect failures in case a check doesn't expose a proper `__version__`, so returning `nil, nil` at line 657 it's like saying "I asked Python, they answered but no version". This is indeed tricky, we should add a comment about this at line 657.